File: Chapter13/ch10-deptservice/src/main/java/org/packt/microservice/core/dao/DepartmentRepository.java
package org.packt.microservice.core.dao;

import java.util.List;

import org.packt.microservice.core.model.data.Department;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DepartmentRepository extends JpaRepository<Department, Integer>{
	
	@Cacheable("departmentCache")
	public List<Department> findByName(String name);
	
	@Cacheable("departmentCache")
	public List<Department> findByDeptid(Integer deptId);

}

File: Chapter13/ch12-kotlin/src/main/kotlin/org/packt/microservice/kotlin/dao/DepartmentRepository.kt
package org.packt.microservice.kotlin.dao

import org.springframework.data.repository.CrudRepository
import org.packt.microservice.kotlin.model.data.Department

interface DepartmentRepository : CrudRepository<Department, Integer>{
	
	fun findByName(name: String): Iterable<Department>
	fun findByDeptid(deptid: Integer): Iterable<Department>
	
}